### PR TITLE
Add security test testing SLM cluster privileges

### PR DESCRIPTION
**Pull request by dakrone**
This adds a test to `PermissionsIT` that uses the `manage_slm` and `read_slm`
cluster privileges.

Relates to #38461

### CODE REVIEW

**Comment by dakrone** [comment]
@elasticmachine run elasticsearch-ci/1

**Comment by dakrone** [comment]
@elasticmachine run elasticsearch-ci/1

File: x-pack/plugin/ilm/qa/with-security/src/test/java/org/elasticsearch/xpack/security/PermissionsIT.java [hunk 1]
```diff
@@ -126,6 +140,95 @@ public void testCanManageIndexWithNoPermissions() throws Exception {
...
+        // Build two high level clients, each using a different user
+        RestClientBuilder builder = RestClient.builder(adminClient().getNodes().toArray(new Node[0]));
+        String token = basicAuthHeaderValue("slm_admin", new SecureString("slm-pass".toCharArray()));
+        configureClient(builder, Settings.builder()
+            .put(ThreadContext.PREFIX + ".Authorization", token)
+            .build());
+        builder.setStrictDeprecationMode(true);
+        final RestHighLevelClient adminHLRC = new RestHighLevelClient(builder);
+
+        builder = RestClient.builder(adminClient().getNodes().toArray(new Node[0]));
```
Comment (gwbrown) [inline x-pack/plugin/ilm/qa/with-security/src/test/java/org/elasticsearch/xpack/security/PermissionsIT.java]: I don't love reusing variables in general (`builder` here, `token` in the next line) but it's not a major objection.

**Review by AthenaEryma — APPROVED** [review]
LGTM, understandable even with my lacking understanding of how Security works. Left one nitpick.

**Comment by dakrone** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by dakrone** [comment]
@elasticmachine update branch

**Pull request merged**